packages/plugins/i18n/server/src/services/localizations.ts

```diff
@@ -1,40 +1,96 @@
-import { isEmpty } from 'lodash/fp';
+import { cloneDeep, isEmpty } from 'lodash/fp';
 
-import { async } from '@strapi/utils';
+import { type Schema } from '@strapi/types';
+import { async, traverseEntity } from '@strapi/utils';
 import { getService } from '../utils';
 
-const isDialectMySQL = () => strapi.db.dialect.client === 'mysql';
+const omitIdAndLocale = (entry?: object | null) => {
+  if (entry && typeof entry === 'object' && 'id' in entry && 'locale' in entry) {
+    const { id, locale, ...rest } = entry;
+    return rest;
+  }
+
+  return entry;
+};
+
+const omitFieldsFromRelations = (data: any, schema: Schema.ContentType) => {
+  return traverseEntity(
+    ({ key, value, attribute }, { set }) => {
+      if (attribute.type === 'relation') {
+        if (Array.isArray(value)) {
+          const newValue = value.map((entry) => {
+            return omitIdAndLocale(entry);
+          });
+
+          set(key, newValue as any);
+        } else if (typeof value === 'object') {
+          set(key, omitIdAndLocale(value) as any);
+        }
+      }
+    },
+    { schema, getModel: strapi.getModel.bind(strapi) },
+    data
+  );
+};
```

Comment:
is it necessary to omit those?

Could we make the relation transformations smart enough , so if its a relation between two i18n content types, it uses the locale of the source entry ?

Reply:
Thanks, done that in [`0739c5e` (#19720)](https://github.com/strapi/strapi/pull/19720/commits/0739c5ed282e615ce70ec9a8550fe943011b3512)